execution_engine/src/storage/global_state/lmdb.rs

```diff
@@ -281,16 +289,68 @@ impl StateProvider for LmdbGlobalState {
         correlation_id: CorrelationId,
         trie_keys: Vec<Digest>,
     ) -> Result<Vec<Digest>, Self::Error> {
-        let txn = self.environment.create_read_txn()?;
-        let missing_descendants =
-            missing_trie_keys::<Key, StoredValue, lmdb::RoTransaction, LmdbTrieStore, Self::Error>(
+        let trie_count = {
+            let digests_without_missing_descendants = self
+                .digests_without_missing_descendants
+                .read()
+                .expect("digest cache read lock");
+            trie_keys
+                .iter()
+                .filter(|digest| !digests_without_missing_descendants.contains(digest))
+                .count()
```

Comment:
We should remove the `trie_keys` without missing descendants even if the count is not 0, so we don't redundantly run `missing_trie_keys` on them.

(Pointing this out again because https://github.com/casper-network/casper-node/pull/2857#discussion_r849432211 is "outdated".)

Reply:
I put my comment on a wrong thread... reposting here:

I've looked at the call sites and we always call this method (`missing_trie_keys`) with just one `Digest`. This means that it's either zero or single trie key we will call the `missing_trie_keys` with.

Reply:
OK, then I guess it's not a blocker, but still…
We could make the function take just a single one as an argument.

Reply:
Didn't want to change an API for a public method as part of a "bug fix". We can ticket that work.